cfgov/v1/signals.py

```diff
@@ -35,3 +38,10 @@ def user_save_callback(sender, **kwargs):
         current_password_history = user.passwordhistoryitem_set.latest()
         if user.password != current_password_history.encrypted_password:
             new_phi(user)
+
+
+def invalidate_post_preview(sender, **kwargs):
+    instance = kwargs['instance']
+    caches['post_preview'].delete(instance.post_preview_cache_key)
+
+page_published.connect(invalidate_post_preview)
```

Comment:
This call will only run on a single server -- the one the publish happens on (content) -- and so won't invalidate the file-based caches on any of the other servers. Unless the servers share some central cache, we need a way to invalidate everywhere, otherwise those other servers will continue to serve stale content.

Reply:
Updated to use a database cache in 574d22f23f47652827a29526a0c697332c2ca5b5